production.py

```diff
@@ -1,38 +1,114 @@
         context={
-            'default_quality_template': Eval('quality_template'),
+            'default_quality_templates': Eval('quality_templates'),
             },
-        depends=['quality_template'])
+        depends=['quality_templates'])
+    time_since_quality_control = fields.DateTime('Time since quality control',
+        readonly=True)
 
-    @fields.depends('product')
-    def on_change_product(self):
-        super(Production, self).on_change_product()
-        if self.product and self.product.template.quality_template:
-            self.quality_template = self.product.template.quality_template
+    def get_quality_templates(self, name):
+        context = Transaction().context
+
+        quality_templates = []
+        if self.product:
+            for quality_template in self.product.template.quality_templates:
+                if quality_template.company.id == context.get('company'):
+                    quality_templates.append(quality_template.id)
+        return quality_templates
 
     @classmethod
     def compute_request(cls, product, warehouse, quantity, date, company):
         "Inherited from stock_supply_production"
         production = super(Production, cls).compute_request(product,
             warehouse, quantity, date, company)
-        if product.template.quality_template:
-            production.quality_template = product.template.quality_template
+        if product.template.quality_templates:
+            production.quality_templates = product.template.quality_templates
         return production
+
+    @classmethod
+    def run(cls, productions):
+        to_save = []
+        for production in productions:
+            production.time_since_quality_control = datetime.now()
+            to_save.append(production)
+        cls.save(to_save)
+        super(Production, cls).run(productions)
+
+    @classmethod
+    def create_quality_tests(cls, productions):
+        pool = Pool()
+        QualityTest = pool.get('quality.test')
+
+        to_save = []
+        for production in productions:
+            for quality_template in production.quality_templates:
+                total_time_since_quality_control = (datetime.now() -
+                    production.time_since_quality_control).seconds/60
+                if (total_time_since_quality_control >=
+                        quality_template.interval):
+                    for quality_contol in range(int(
+                            total_time_since_quality_control/
+                            quality_template.interval)):
+                        test = QualityTest()
+                        test.document = production
+                        test.templates = [quality_template.quality_template]
+                        test.company = quality_template.company
+                        to_save.append(test)
+
+                    production.time_since_quality_control = datetime.now()
+                    production.save()
```

Comment:
You should save all productions at once at the end of the loop:

cls.save(productions)